log/sentry.go

```diff
@@ -67,16 +70,24 @@ type SentryConfig struct {
 // this function also sets hostname tag to the value reported by the kernel
 // (when cfg.ServerName is empty server_name tag will be the hostname).
 func InitSentry(cfg SentryConfig) (io.Closer, error) {
-	var sampleRate float64 = 1
+	var (
+		sampleRate float64        = 1
+		beforeSend beforeSendFunc = dropStackTraceFrame
+	)
 	if cfg.SampleRate != nil && *cfg.SampleRate >= 0 && *cfg.SampleRate <= 1 {
 		sampleRate = *cfg.SampleRate
 	}
+	if cfg.BeforeSend != nil {
+		beforeSend = cfg.BeforeSend
+	}
```

Comment:
since `sentry.ClientOptions.BeforeSend` can only take a single function, what if the user want to override `BeforeSend` to do something else, but they also want to keep this default behavior that they want to drop the first frame?

I think we should export `dropStackTraceFrame` so callers can call it in their implementation, and probably also provide a way to merge in multiple `BeforeSend` functions.

Reply:
I wonder if there is a way with less additional types to deal with. The way I saw this implemented in the Bugsnag library is to have an option called skip levels that just defines a number of frames to skip.

What if we export the whole configuration but we also apply it but wrap the `BeforeSend` method ourselves? Too much magic?

Reply:
what I meant is just to export `dropStackTraceFrame` to `DropStackTraceFrame`, so if the user want to do something else they don't have to reimplement `dropStackTraceFrame`:

```
func myBeforeSend(event *sentry.Event, hint *sentry.EventHint) *sentry.Event {
  event = log.DropStackTraceFrame(event, hint)
  // do extra work
}
```

Reply:
you can kinda already do

orig := cfg.BeforeSend
cfg.BeforeSend = func(...) {
  orig()
  ...
}

and you could make a helper to do that